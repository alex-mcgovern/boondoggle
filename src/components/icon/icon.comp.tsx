import { extractAtomsFromProps } from "@dessert-box/core";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import clsx from "clsx";

import { getSprinkles } from "../../styles/utils/get_sprinkles.css";
import * as styles from "./icon.styles.css";

import type { SprinklesArgs } from "../../styles/utils/get_sprinkles.css";
import type { IconProp } from "@fortawesome/fontawesome-svg-core";
import type { FontAwesomeIconProps } from "@fortawesome/react-fontawesome";

export type IconProps = SprinklesArgs &
  Pick<FontAwesomeIconProps, "size" | "spin"> & {
    className?: string;
    icon: IconProp;
  };

export function Icon({ icon, className: userClassName, ...rest }: IconProps) {
  const { atomProps, otherProps } = extractAtomsFromProps(rest, getSprinkles);

  return (
    <FontAwesomeIcon
      className={clsx(
        styles.icon,
        userClassName,
        getSprinkles({ ...atomProps, flexShrink: "0" })
      )}
      icon={icon}
      {...otherProps}
    />
  );
}
