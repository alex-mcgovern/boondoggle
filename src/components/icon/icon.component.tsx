import { extractAtomsFromProps } from "@dessert-box/core";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import clsx from "clsx";
import React from "react";

import { getSprinkles } from "../../styles/utils/get_sprinkles.css";
import * as styles from "./icon.styles.css";

import type {
  SprinklesArgs,
  SprinklesColor,
  SprinklesMargin,
  SprinklesPadding,
  SprinklesSize,
} from "../../styles/utils/get_sprinkles.css";
import type { IconProp } from "@fortawesome/fontawesome-svg-core";
import type { FontAwesomeIconProps } from "@fortawesome/react-fontawesome";

export interface IconProps
  extends SprinklesMargin,
    SprinklesPadding,
    SprinklesColor,
    SprinklesSize,
    Pick<SprinklesArgs, "display">,
    Pick<FontAwesomeIconProps, "size" | "spin"> {
  icon: IconProp;
  className?: string;
}

export function Icon({ icon, className: userClassName, ...rest }: IconProps) {
  const { atomProps, otherProps } = extractAtomsFromProps(rest, getSprinkles);

  return (
    <FontAwesomeIcon
      icon={icon}
      className={clsx(
        styles.icon,
        userClassName,
        getSprinkles({ ...atomProps, flexShrink: "0" })
      )}
      {...otherProps}
    />
  );
}
