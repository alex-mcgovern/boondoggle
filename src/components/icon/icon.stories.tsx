import { faShapes } from "@fortawesome/free-solid-svg-icons";
import React from "react";

import { Icon as StoryComponent } from ".";

import type { IconProps as StoryComponentProps } from ".";
import type { StoryFn, StoryObj } from "@storybook/react";

export default {
  title: "React components/Icon",
  component: StoryComponent,
};

const Template: StoryFn<StoryComponentProps> = ({
  ...rest
}: StoryComponentProps) => {
  return <StoryComponent {...rest} />;
};

export const Default: StoryObj<StoryComponentProps> = {
  render: Template,
  name: "default",
  args: {
    icon: faShapes,
  },
};
