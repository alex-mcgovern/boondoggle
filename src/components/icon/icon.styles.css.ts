import { style } from "@vanilla-extract/css";

import { vars } from "../../styles/theme.css";
import { createAccessibleTransition } from "../../styles/utils/create_accessible_transition.css";

export const icon = style([
  createAccessibleTransition({
    transition: `transform ${vars.transitionDuration.short} ease`,
  }),
]);
