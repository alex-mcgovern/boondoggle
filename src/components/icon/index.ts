export { Icon } from "./icon.component";
export type { IconProps } from "./icon.component";
