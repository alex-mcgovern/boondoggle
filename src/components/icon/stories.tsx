import { faShapes } from "@fortawesome/pro-solid-svg-icons";

import { Icon as StoryComp } from ".";

import type { StoryFn, StoryObj } from "@storybook/react";
import type { IconProps as StoryCompProps } from ".";

export default {
	component: StoryComp,
	title: "Components/Icon",
};

const Template: StoryFn<StoryCompProps> = ({ ...rest }: StoryCompProps) => {
	return <StoryComp {...rest} />;
};

export const Default: StoryObj<StoryCompProps> = {
	args: {
		icon: faShapes,
	},
	render: Template,
};
