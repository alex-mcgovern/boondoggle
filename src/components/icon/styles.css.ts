import { style } from "@vanilla-extract/css";
import { createAccessibleTransition, vars } from "../../index.css";

export const icon = style([
	createAccessibleTransition({
		transition: `transform color ${vars.transitionDuration.short} ease`,
	}),
]);
