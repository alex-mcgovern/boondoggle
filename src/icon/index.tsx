import type { IconProp } from "@fortawesome/fontawesome-svg-core";
import type { FontAwesomeIconProps } from "@fortawesome/react-fontawesome";
import type { Color } from "react-aria-components";

import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import clsx from "clsx";

import "./styles.css";

/**
 * A wrapper around the FontAwesome icon component.
 */
export function Icon({
    className,
    color,
    icon,
    ...rest
}: FontAwesomeIconProps & {
    /**
     * An optional class name to apply to the icon.
     */
    className?: string;
    /**
     * The color of the icon.
     */
    color?: Color;

    /**
     * The FontAwesome icon definition to render.
     */
    icon: IconProp;
}) {
    return (
        <FontAwesomeIcon
            className={clsx("icon", className, color)}
            icon={icon}
            {...rest}
        />
    );
}
