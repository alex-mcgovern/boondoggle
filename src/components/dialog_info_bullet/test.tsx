/**
 * @jest-environment jsdom
 */
import { faShapes } from "@fortawesome/sharp-regular-svg-icons";
import { render } from "@testing-library/react";

import { DialogInfoBullet } from ".";
import { LOREM } from "../../../mocks/LOREM.mock";
import { Icon } from "../icon";

import type { DialogInfoBulletProps } from ".";

const renderComponent = ({ ...props }: DialogInfoBulletProps) =>
    {return render(<DialogInfoBullet {...props} />)};

const PROPS: DialogInfoBulletProps = {
    slotLeft: [
        <div>
            <Icon icon={faShapes} />
        </div>,
    ],
    slotRight: [
        <div>
            <Icon icon={faShapes} />
        </div>,
    ],
    text: LOREM.text_xxs,
    title: LOREM.title_xl,
};

describe("<DialogInfoBullet />", () => {
    describe("Basic smoke tests", () => {
        test("should render without throwing", () => {
            const { container } = renderComponent(PROPS);

            expect(container).not.toBeNull();
        });
    });
});
