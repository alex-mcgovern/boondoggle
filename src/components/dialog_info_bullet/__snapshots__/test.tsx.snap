// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DialogInfoBullet /> Basic smoke tests should match snapshot 1`] = `
<div>
  <div
    class="get_sprinkles_alignItems_center__eiij7b2 get_sprinkles_display_flex_mobile__eiij7bb0 get_sprinkles_gap_space_4_mobile__eiij7bco get_sprinkles_marginBottom_space_4__eiij7b4q"
  >
    <div
      class="get_sprinkles_flexShrink_0__eiij7b1w get_sprinkles_width_space_4_mobile__eiij7bp3"
    >
      <div>
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-shapes styles_icon__1ej8lf00 get_sprinkles_color_currentColor__eiij7b1o get_sprinkles_flexShrink_0__eiij7b1w"
          data-icon="shapes"
          data-prefix="fasr"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M160 224h55.3H360.7 416l-27.4-48L315.6 48.4 288 0 260.4 48.4 187.4 176 160 224zM288 96.7L333.3 176H242.7L288 96.7zM128 304a80 80 0 1 1 0 160 80 80 0 1 1 0-160zm0 208a128 128 0 1 0 0-256 128 128 0 1 0 0 256zM464 320V448H336V320H464zM336 272H288v48V448v48h48H464h48V448 320 272H464 336z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
    <div>
      <div
        class="get_sprinkles_fontWeight_semibold__eiij7b2i"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed euismod, nunc vel tincidunt lacinia, nunc nisl aliquam nisl, eget aliquam nisl.
      </div>
      <div>
        Lorem ipsum
      </div>
    </div>
    <div
      class="get_sprinkles_flexShrink_0__eiij7b1w get_sprinkles_width_space_4_mobile__eiij7bp3"
    >
      <div>
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-shapes styles_icon__1ej8lf00 get_sprinkles_color_currentColor__eiij7b1o get_sprinkles_flexShrink_0__eiij7b1w"
          data-icon="shapes"
          data-prefix="fasr"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M160 224h55.3H360.7 416l-27.4-48L315.6 48.4 288 0 260.4 48.4 187.4 176 160 224zM288 96.7L333.3 176H242.7L288 96.7zM128 304a80 80 0 1 1 0 160 80 80 0 1 1 0-160zm0 208a128 128 0 1 0 0-256 128 128 0 1 0 0 256zM464 320V448H336V320H464zM336 272H288v48V448v48h48H464h48V448 320 272H464 336z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
